pkg/build/build.go

```diff
@@ -350,6 +363,74 @@ func pullImage(ctx *buildContext, image string) (id string, err error) {
 	return pushTarget.Context().Digest(digest.String()).String(), nil
 }
 
+func isImageRemote(ctx *buildContext, image string) (bool, error) {
+	ref, err := images2.ParseReferenceNoDefault(image)
+	if err != nil {
+		return false, nil
+	}
+
+	_, err = remote.Index(ref, ctx.remoteOpts...)
+	if err != nil {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func resolveLocalImage(ctx *buildContext, image string) (string, error) {
+	// We need to be careful to make sure we only list ImageInstances in the namespace to which the user has access.
+	// We can determine the namespace based on the pushRepo, which is in the format <registry>/acorn/<namespace>.
+	// To be very safe, verify the structure of pushRepo before getting the namespace from it.
+	if len(strings.Split(ctx.pushRepo, "/")) != 3 || !strings.HasPrefix(ctx.pushRepo, "127.0.0.1:") || strings.Split(ctx.pushRepo, "/")[1] != "acorn" {
+		return "", fmt.Errorf("invalid pushRepo: %s", ctx.pushRepo)
+	}
+
+	namespace := strings.Split(ctx.pushRepo, "/")[2]
```

Comment:
This is the part of this that I dislike the most. I tried to put in enough safeguards such that, if in the future someone refactors how `pushRepo` works, it should be caught here and errors thrown instead of accidentally granting image access across projects.

One possible way around this would be to store another piece of information besides pushRepo in the Token, and that would simply be the namespace (project name) itself. That would be duplication of information for now, since that info is extractable from the pushRepo, but then it would at least be more explicit that that is something we are relying on. Thoughts?

Reply:
Can this [`token.Build.Namespace`](https://github.com/acorn-io/acorn/blob/dd4f1c7c8b6add7c9ecbfc2afe5e10bac0be5964/pkg/buildserver/server.go#L125) be added to the [`buildContext`](https://github.com/acorn-io/acorn/blob/959469c916c908502448dab058258701c99c506c/pkg/build/build.go#L38) to make this simpler?

Reply:
Oooh, very nice. I will update it to use that